WalletWasabi.Fluent/AddWallet/Common/EnterPasswordViewModel.cs

```diff
@@ -0,0 +1,83 @@
+using ReactiveUI;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using WalletWasabi.Fluent.ViewModels.Dialogs;
+using WalletWasabi.Gui.Validation;
+using WalletWasabi.Models;
+using WalletWasabi.Userfacing;
+
+namespace WalletWasabi.Fluent.AddWallet.Common
+{
+	public class EnterPasswordViewModel : DialogViewModelBase<string>
+	{
+		private IScreen _screen;
+		private string _password;
+		private string _confirmPassword;
+
+		public EnterPasswordViewModel(IScreen screen)
+		{
+			_screen = screen;
+
+			this.ValidateProperty(x => x.Password, ValidatePassword);
+			this.ValidateProperty(x => x.ConfirmPassword, ValidateConfirmPassword);
+
+			var continueCommandCanExecute = this.WhenAnyValue(
+				x => x.Password,
+				x => x.ConfirmPassword,
+				(password, confirmPassword) =>
+				{
+					// This will fire validations before return canExecute value.
+					this.RaisePropertyChanged(nameof(Password));
+					this.RaisePropertyChanged(nameof(ConfirmPassword));
+
+					return (string.IsNullOrEmpty(password) && string.IsNullOrEmpty(confirmPassword)) || (!string.IsNullOrEmpty(password) && !string.IsNullOrEmpty(confirmPassword) && !Validations.Any);
+				})
+				.ObserveOn(RxApp.MainThreadScheduler);
```

Comment:
there is no subscribe is this code even being used?

Reply:
Yes, this is the continue command's CanExecute condition.